x-pack/test/functional/page_objects/dataset_quality.ts

```diff
@@ -470,12 +478,7 @@ async function parseDatasetTable(tableWrapper: WebElementWrapper, columnNamesOrI
         cellContentElements: cellContentWrappers,
         getSortDirection,
         sort: async (sortDirection: 'ascending' | 'descending') => {
-          if ((await getSortDirection()) !== sortDirection) {
-            await headerElement.click();
-          }
-
-          // Sorting twice if the sort was in neutral state
-          if ((await getSortDirection()) !== sortDirection) {
+          while ((await getSortDirection()) !== sortDirection) {
```

Comment:
If `headerElement.click()` fails for an unprecedented reason, how long `while` will keep rolling?

Maybe use `retry`, `tryForTimeout` or limit the max iterations.

Reply:
I too had this feeling when writing that what if. But i couldn't find of a situation where this would happen. But as a precautionary, i would wrap it inside a `tryForTime` 👍🏼